### PR TITLE
dev-games/openscenegraph: version bump to 3.5.5

### PR DESCRIPTION
**Pull request by Hummer12007**
dev-games/osgQt: new package (Qt support split off osg in a separate package).
Keeping old maintainer (as it's a module of the same package). Done in one commit, as separating would lead to breakage after the first commit, but before the second.

### CODE REVIEW

**Comment by Hummer12007** [comment]
@reavertm

**Comment by a17r** [comment]
Why the CMakeLists.txt patch, did you take it from upstream?

**Comment by Hummer12007** [comment]
Took the one from 3.5.1 and made it work with 3.5.5.

**Comment by Polynomial-C** [comment]
There's currently no _dev-games/simgear_ version available that successfully compiles with anything newer than openscenegraph-3.5.1
So this bump either needs to be p.masked or simgear's dependency needs to be adjusted to exclude any newer openscenegraph versions until there's a simgear release that works with latest openscenegraph.

**Comment by Hummer12007** [comment]
Hmm, indeed, added a blocker to that package.

File: dev-games/simgear/simgear-2017.1.3.ebuild [hunk 1]
```diff
@@ -14,9 +14,11 @@ KEYWORDS="~amd64 ~x86"
 SLOT="0"
 IUSE="+dns debug subversion test"
 
+#NOTE: current head works with osg-3.5.5
 COMMON_DEPEND="
 	dev-libs/expat
 	>=dev-games/openscenegraph-3.2.0
+	!>=dev-games/openscenegraph-3.5.2
```
Comment (mgorny) [inline dev-games/simgear/simgear-2017.1.3.ebuild]: Do not use a blocker, just use `<` dep.

File: dev-games/osgQt/osgQt-3.5.5.ebuild [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit cmake-utils vcs-snapshot
```
Comment (mgorny) [inline dev-games/osgQt/osgQt-3.5.5.ebuild]: Do you actually need vcs-snapshot here? I think github gets the directory name right when tag is named after version.

File: dev-games/openscenegraph/openscenegraph-3.5.5.ebuild [hunk 3]
```diff
@@ -0,0 +1,148 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+WX_GTK_VER="3.0"
+
+inherit cmake-utils flag-o-matic wxwidgets vcs-snapshot
```
Comment (mgorny) [inline dev-games/openscenegraph/openscenegraph-3.5.5.ebuild]: I think vcs-snapshot is not needed here. Especially that you override S.

File: dev-games/simgear/simgear-2017.1.3.ebuild (hunk 1 above)
Comment (Hummer12007) [inline dev-games/simgear/simgear-2017.1.3.ebuild]: Fixed

File: dev-games/osgQt/osgQt-3.5.5.ebuild (hunk 2 above)
Comment (Hummer12007) [inline dev-games/osgQt/osgQt-3.5.5.ebuild]: Oh, you're right, it does. vcs-snapshot is redundant here indeed.

File: dev-games/openscenegraph/openscenegraph-3.5.5.ebuild (hunk 3 above)
Comment (Hummer12007) [inline dev-games/openscenegraph/openscenegraph-3.5.5.ebuild]: It is

**Comment by Hummer12007** [comment]
Addressed everything

**Comment by reavertm** [comment]
Thanks for the patch.
Some issues I had to address before merging though:
- wrong installation location for compiled examples in osgQt (share/OpenSceneGraph but should be bin)
- doc USE flag doesn't work for osgQt (incomplete buildsystem)
- unnecessary qt5 USE flag in osgQt
- missing optional dep on qtwebkit5 in osgQt (examples)
I also did the following changes:
- packages masked instead of adding block (there are other consumers, not just simgear)
- added proper subslot tracking (by SOVERSION)
- renamed osgQt to openscenegraph-qt

**Comment by Hummer12007** [comment]
@reavertm i've left the qt flag in openscenegraph for a reason: some rdeps (which work with openscenegraph-3.5.5 like openmw) depend on it.

**Comment by reavertm** [comment]
I removed it on purpose to trigger more conscious adaptation and migration to split Qt ebuild. osg-3.5.5 is not drop-in replacement due to API has changes anyway.

**Pull request closed without merging**